ansible/server-state-playbooks/pg-int-testing.yml

```diff
@@ -0,0 +1,113 @@
+# Install PostgreSQL and prepare the OME (UoD/SLS) prerequisites
+
+- hosts: pg-int-testing.openmicroscopy.org
+  pre_tasks:
+    - name: Install open-vm-tools if system is a VMware vm
+      become: yes
+      yum:
+        name: open-vm-tools
+        state: latest
+      when: >
+           ((ansible_virtualization_type is defined)
+           and (ansible_virtualization_type == "VMware"))
+
+    # Perhaps alter the role at https://github.com/openmicroscopy/ansible-role-lvm-partition/
+    # to make some of the variables non-required.
+    - name: Resize root FS without altering mount options
+      tags: lvm
+      become: yes
+      lvol:
+        lv: root
+        vg: VolGroup00
+        size: "{{ provision_root_lvsize }}"
+        shrink: no
+
+  roles:
+
+    # Disk Layout - PostgreSQL | data dir on separate VG (SSD) 
+    - role: openmicroscopy.lvm-partition
+      tags: lvm
+      lvm_lvname: pgdata
+      lvm_vgname: "{{ provision_postgres_vgname }}"
+      lvm_lvmount: /var/lib/pgsql
+      lvm_lvsize: "{{ provision_postgres_lvsize }}"
+      lvm_lvfilesystem: "{{ filesystem }}"
+      lvm_shrink: False
+
+    # Checking `is server deployed in Dundee/SLS` is checking for the SLS nameservers.
+    - role: openmicroscopy.system-monitor-agent
+      tags: monitoring
+      when: "'10.1.255.216' in ansible_dns.nameservers"
+
+    - role: openmicroscopy.basedeps
+
+    - role: openmicroscopy.postgresql
+      postgresql_install_server: True
+  
+  post_tasks:
+
+    - name: Check_MK postgres plugin | check for plugin existence
+      tags: monitoring
+      stat:
+        path: "{{ check_mk_agent_plugin_path }}/mk_postgres"
+      register: check_mk_postgres_plugin_st
+
+    - name: Check_MK postgres plugin | activate the plugin
+      tags: monitoring
+      become: yes
+      command: cp "{{ check_mk_agent_plugin_path }}/mk_postgres" /usr/share/check-mk-agent/plugins/ creates=/usr/share/check-mk-agent/plugins/mk_postgres
+      when: check_mk_postgres_plugin_st.stat.exists
+
+    - name: Check_MK logwatch plugin | check for plugin existence
+      tags: monitoring
+      stat:
+        path: "{{ check_mk_agent_plugin_path }}/mk_logwatch"
+      register: check_mk_logwatch_plugin_st
+
+    - name: Check_MK logwatch plugin | activate the plugin
+      tags: monitoring
+      become: yes
+      command: cp "{{ check_mk_agent_plugin_path }}/mk_logwatch" /usr/share/check-mk-agent/plugins/ creates=/usr/share/check-mk-agent/plugins/mk_logwatch
+      when: check_mk_logwatch_plugin_st.stat.exists
```

Comment:
Why would these plugins not exist if?

Reply:
To stop the whole playbook failing when
* The agent install didn't work for some reason
* If there's a change to the packaging of the agent

Some of the earlier systems don't have all the plugins, so I think point 2 may be relevant in that case.

Reply:
I can also see that bit of code being a function or put into the upstream role which accepts `plugin-name`, but that's for a refactor / ☔️  day

Reply:
This is a new playbook though, so shouldn't everything be working?

Reply:
It'd take longer to deploy a system if one were to whittle down working/tested code to only that which "should" be necessary every time. Every change introduces the chance of error, right?

Reply:
Ansible should give you a well-defined system. If there's uncertainty due to interactions between multiple roles or playbooks this should be highlighted as an issue (and we can decide whether to treat it as a known issue or a bug which needs fixing, as it may cause problems when someone else deploys a new system.